src/components/QuestionBox.vue

```diff
@@ -46,90 +34,59 @@
 <script>
 import _ from 'lodash'
 
-
 export default {
   props: {
-    currentQuestion: Object,
-    nextQuest: Function,
-    increment : Function,
-    numTotal : Number
+    question: Object,
   },
   data(){
     return {
-      selectedAnswer : null,
-      shuffledAnswers : [],
-      correctIndex : null,
-      answered : false,
-      theQuestion : ''
-    }
+      selectedAnswer : null 
+  }
   },
-  watch: {
-    currentQuestion:{
-      immediate: true,
-      handler() {
-        this.selectedAnswer = null;
-        this.answered = false;
-        this.shuffleAnswers();
-        this.theQuestion = this.decodeHTML(this.currentQuestion.question);
-      }
-      
+  computed:{
+    shuffledAnswers(){
+      let shuffledAnswers = []
+      shuffledAnswers = _.shuffle(this.question.answers)
+      return shuffledAnswers
+    },
+    correctIndex(){
+      let correctIndex = this.shuffledAnswers.findIndex(x => x.id ===1)
+      return correctIndex
     }
   },
   methods:{
-    decodeHTML(html){
-      let txt = document.createElement("textarea");
-      txt.innerHTML = html;
-      return txt.value;
+    answerClass(index){
+      let answerClass = ''
+      if(index === this.correctIndex && this.selectedAnswer !== null){
+        answerClass = 'correctAnswer'
+      }else if(index === this.selectedAnswer){
+        answerClass = 'incorrectAnswer'
+      }
+      return answerClass
     },
     selectAnswer(index){
       this.selectedAnswer = index
-      this.updateResults()
-    },
-    shuffleAnswers(){
-      let answers = [...this.currentQuestion.incorrect_answers, this.currentQuestion.correct_answer]
-      this.shuffledAnswers = _.shuffle(answers)
-      this.correctIndex = this.shuffledAnswers.indexOf(this.currentQuestion.correct_answer)
-      this.shuffledAnswers = this.shuffledAnswers.map(reponse => {
-        return this.decodeHTML(reponse)
-      }) 
-
+      this.isCorrect(index)
     },
-    updateResults(){
-      let isCorrect = false 
-
-      if(this.selectedAnswer === this.correctIndex){
+    isCorrect(index){
```

Comment:
bizarre cette méthode, ce code pourrait se trouver dans `selectAnswer()`

Reply:
separation of concerns ? ça me paraissait plus propre/explicite d'avoir une fonction spécifique pour la vérification de la réponse.

Reply:
Alors ta méthode « isCorrect » ne devrait pas muter ton state ;)

Reply:
bah elle ne le mute pas, puisqu'elle modifie seulement une variable dans son propre scope. Je crois qu'il y a un truc que je pige pas.